platform/openide.filesystems/src/org/openide/filesystems/Ordering.java

```diff
@@ -158,7 +164,8 @@ public int compareTo(ChildAndPosition o) {
                 previousChild = subsequentChild;
             }
         }
-        if (logWarnings && /* #201893*/ !parent.getPath().matches("Projects/.+/Lookup") && !childrenByPosition.isEmpty() && childrenByPosition.size() < children.size()) {
+        if (logWarnings && /* #201893*/ !parent.getPath().matches("Projects/.+/Lookup") && !childrenByPosition.isEmpty() && childrenByPosition.size() < children.size() && 
```

Comment:
The registration folder typically contains `.instance` files and subfolders (categories, ...); while if one `.instance` is position-ed, all `.instance`s should define positions (this is the point of the test).
It seems that it is NOT necessary to require order on files+folders, but just order within the category.